plugins/nvme/fu-nvme-device.c

```diff
@@ -267,6 +267,29 @@ fu_nvme_device_dump(const gchar *title, const guint8 *buf, gsize sz)
 	g_print("\n");
 }
 
+/* Returns %TRUE if device is an NVMe-over-Fabrics device */
+static gboolean
+fu_nvme_device_is_fabrics(FuDevice *device, GError **error)
+{
+	g_autoptr(GUdevDevice) device_tmp = NULL;
+	GUdevDevice *gdev;
+
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
+
+	gdev = fu_udev_device_get_dev(FU_UDEV_DEVICE(device));
+
+	device_tmp = g_udev_device_get_parent_with_subsystem(gdev, "nvme-fabrics", NULL);
+	if (device_tmp != NULL) {
+		g_set_error(error,
+			    FWUPD_ERROR,
+			    FWUPD_ERROR_NOT_SUPPORTED,
+			    "device is NVMe-over-Fabrics");
+		return TRUE;
```

Comment:
It's unusual for return true on the failure scenario in this project. Could you instead swap the logic around?

Maybe just match for pci in this new function and if it's anything else fail?

Reply:
Thanks. I force pushed an amend with those changes. Please review.